main/src/cgeo/geocaching/ui/dialog/CoordinatesCalculateDialog.java

```diff
@@ -1011,21 +1088,40 @@ private void updateResult() {
                     lon = getLonResult();
                     break;
             }
+
+            latFormatted = formatCoordinateString(lat, latLeadingZerosAdded, invalidColour);
+            lonFormatted = formatCoordinateString(lon, lonLeadingZerosAdded, invalidColour);
         } else {
             resultColour = invalidColour;
             doneIcon = stateSaved ? R.drawable.ic_menu_saved : R.drawable.ic_menu_save;
 
-            lat = getLatResult();
-            lon = getLonResult();
+            latFormatted = new SpannableString(getLatResult());
+            lonFormatted = new SpannableString(getLonResult());
         }
 
         doneButton.setImageResource(doneIcon);
-        tLatResult.setText(lat);
-        tLonResult.setText(lon);
+        tLatResult.setText(latFormatted);
+        tLonResult.setText(lonFormatted);
         tLatResult.setTextColor(resultColour);
         tLonResult.setTextColor(resultColour);
     }
 
+    private Spannable formatCoordinateString(String coordinateString, int leadingZeros, int paddingColour)

```

Comment:
I don't want to be picky, but my IDE says final modifiers are missing here. And the opening { is on the next line, should be on this.